source/extensions/filters/http/compressor/gzip/config.cc

```diff
@@ -0,0 +1,103 @@
+
+Envoy::Compressor::ZlibCompressorImpl::CompressionLevel GzipCompressorFactory::compressionLevelEnum(
+    envoy::extensions::filters::http::compressor::gzip::v3::Gzip::CompressionLevel
+        compression_level) {
+  switch (compression_level) {
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::BEST_SPEED:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionLevel::Speed;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::COMPRESSION_LEVEL_2:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionLevel::Level2;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::COMPRESSION_LEVEL_3:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionLevel::Level3;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::COMPRESSION_LEVEL_4:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionLevel::Level4;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::COMPRESSION_LEVEL_5:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionLevel::Level5;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::COMPRESSION_LEVEL_6:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionLevel::Level6;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::COMPRESSION_LEVEL_7:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionLevel::Level7;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::COMPRESSION_LEVEL_8:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionLevel::Level8;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::BEST_COMPRESSION:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionLevel::Best;
+  default:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionLevel::Standard;
+  }
+}
+
+Envoy::Compressor::ZlibCompressorImpl::CompressionStrategy
+GzipCompressorFactory::compressionStrategyEnum(
+    envoy::extensions::filters::http::compressor::gzip::v3::Gzip::CompressionStrategy
+        compression_strategy) {
+  switch (compression_strategy) {
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::FILTERED:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionStrategy::Filtered;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::FIXED:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionStrategy::Fixed;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::HUFFMAN_ONLY:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionStrategy::Huffman;
+  case envoy::extensions::filters::http::compressor::gzip::v3::Gzip::RLE:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionStrategy::Rle;
+  default:
+    return Envoy::Compressor::ZlibCompressorImpl::CompressionStrategy::Standard;
+  }
+}
+
+uint64_t GzipCompressorFactory::memoryLevelUint(Protobuf::uint32 level) {
```

Comment:
We have utilities to unwrap values with a default `PROTOBUF_GET_WRAPPED_OR_DEFAULT`